tutorials/metrics.md

```diff
@@ -0,0 +1,39 @@
+# CRI-O Metrics
+
+To enable the [Prometheus][0] metrics exporter for CRI-O, either start `crio`
+with `--metrics-enable` or set the corresponding option in
+`/etc/crio/crio.conf`:
+
+```toml
+[crio.metrics]
+enable_metrics = true
+```
+
+The metrics endpoint serves per default on port `9090`. This can be changed via
+the `--metrics-port` command line argument or via the configuration file, too:
+
+```toml
+metrics_port = 9090
+```
+
+If CRI-O runs with enabled metrics, then this can be verified by querying the
+endpoint manually via [curl][1].
+
+```bash
+> curl localhost:9090/metrics
+…
+```
+
+## Available Metrics
+
+Beside the [default golang based metrics][2], CRI-O provides the following additional metrics too:
+
+| Metric Key                             | Possible Labels                                                                                                                                                                                                                                                                                                                                                                                                                                                                            | Type    | Purpose                                                                  |
+| -------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ | ------- | ------------------------------------------------------------------------ |
+| `crio_operations`                      | every CRI-O RPC name: `Attach`, `ContainerStats`, `ContainerStatus`, `CreateContainer`, `Exec`, `ExecSync`, `ImageFsInfo`, `ImageStatus`, `ListContainerStats`, `ListContainers`, `ListImages`, `ListPodSandbox`, `PodSandboxStatus`, `PortForward`, `PullImage`, `RemoveContainer`, `RemoveImage`, `RemovePodSandbox`, `ReopenContainerLog`, `RunPodSandbox`, `StartContainer`, `Status`, `StopContainer`, `StopPodSandbox`, `UpdateContainerResources`, `UpdateRuntimeConfig`, `Version` | Counter | Cumulative number of CRI-O operations by operation type.                 |
```

Comment:
Side question.  I just went looking quickly, are these RPC's documented anywhere?  It seems like they should have a page of their own listing each one with a brief description of how to call it and what it does.

Reply:
Yes, I added a link below the table.